modules/userId/index.js

```diff
@@ -333,45 +333,46 @@ function initSubmodules(submodules, consentData) {
       if (storedId) {
         // cache decoded value (this is copied to every adUnit bid)
         submodule.idObj = submodule.submodule.decode(storedId);
-      }
-      let refreshNeeded = false;
-      if (typeof submodule.config.storage.refreshInSeconds === 'number') {
-        const storedDate = new Date(getStoredValue(submodule.config.storage, 'last'));
-        refreshNeeded = storedDate && (Date.now() - storedDate.getTime() > submodule.config.storage.refreshInSeconds * 1000);
-      }
-      if (!storedId || refreshNeeded) {
-        // getId will return user id data or a function that will load the data
-        const getIdResult = submodule.submodule.getId(submodule.config.params, consentData, storedId);
 
-        // If the getId result has a type of function, it is asynchronous and cannot be called until later
-        if (typeof getIdResult === 'function') {
-          submodule.callback = getIdResult;
-        } else if (getIdResult) {
-          // A getId result that is not a function is assumed to be valid user id data, which should be saved to users local storage or cookies
-          setStoredValue(submodule.config.storage, getIdResult);
-          // cache decoded value (this is copied to every adUnit bid)
-          submodule.idObj = submodule.submodule.decode(getIdResult);
+        // Refresh functionality
+        // if an idSystem defines refreshId() - always call it
+        // if storage.refreshInSeconds also is defined then only call refreshId when the age of the stored value > refeshInSeconds
+        const refreshInSeconds = submodule.config.storage.refreshInSeconds;
+        if (utils.isFn(submodule.submodule.refreshId)) {
+          let shouldRefresh = true;
+          if (utils.isNumber(refreshInSeconds)) {
+            const storedDate = new Date(getStoredValue(submodule.config.storage, 'last'));
+            shouldRefresh = (storedDate && (Date.now() - storedDate.getTime() > refreshInSeconds * 1000));
```

Comment:
Using an explicit if is cleaner for this one. You have unit the var with true. if (!storedDate || Date.now() - storedDate <= refreshInSeconds * 1000) shouldRefresh = false;